src/main/host/syscall/handler/mman.rs

```diff
@@ -239,16 +240,26 @@ impl SyscallHandler {
         );
 
         if matches!(mmap_result, Err(SyscallError::Native)) {
-            let (ctx, thread) = ctx.objs.split_thread();
-            return Ok(thread.native_mmap(
-                &ctx,
+            // TODO: We already made the native syscall above when calling the memory manager's
+            // do_mmap(). Can this really return `SyscallError::Native`, and does trying the native
+            // syscall again really do anything for us?
+            let (process_ctx, thread) = ctx.objs.split_thread();
+            let mmap_result = thread.native_mmap(
+                &process_ctx,
                 addr,
                 len,
                 prot,
                 flags,
                 plugin_fd.unwrap_or(-1),
                 offset,
-            )?);
+            );
```

Comment:
Do the tests still pass if we remove the extra native syscall try?

Reply:
I removed this code and added an assertion that the result is not `SyscallError::Native`. I'll wait to get your approval before I merge.